go/logic/topology_recovery.go

```diff
@@ -1346,11 +1358,16 @@ func CheckAndRecover(specificInstance *inst.InstanceKey, candidateInstanceKey *i
 			// force mode. Keep it synchronuous
 			var topologyRecovery *TopologyRecovery
 			recoveryAttempted, topologyRecovery, err = executeCheckAndRecoverFunction(analysisEntry, candidateInstanceKey, true, skipProcesses)
+			log.Errore(err)
 			if topologyRecovery != nil {
 				promotedReplicaKey = topologyRecovery.SuccessorKey
 			}
 		} else {
-			go executeCheckAndRecoverFunction(analysisEntry, candidateInstanceKey, false, skipProcesses)
+			analysisEntry := analysisEntry
```

Comment:
I believe the proper behavior here would be to pass `analysisEntry` into the goroutine:

``` 
go func(analysisEntry inst.ReplicationAnalysis) {
    // call things in here
}(analysisEntry)
```